### ISSUE TITLE
[BP-568] Having creditcards on and IDEAL off breaks the checkout as of 1.26.0

### ISSUE THREAD

**Issue by wouterfolkertsma** [buckaroo-it/Magento2#56]
Hello everyone,

While deploying a shop for a customer we came across the following bug.
In the following file, the method-renderer for the credit cards tries to fetch a configuration setting for IDEAL from the window.checkoutconfig.

https://github.com/buckaroo-it/Magento2/blob/master/view/frontend/web/js/view/payment/method-renderer/creditcard.js 
On line 78 you see the following: 
`this.selectionType  = window.checkoutConfig.payment.buckaroo.ideal.selectionType;`

When IDEAL is turned off as a payment method, `window.checkoutConfig.payment.buckaroo.ideal` will not be available, and thus the application will throw the following error:

`cannot read "selectionType" of undefined`.

I have been able to reproduce this in a live environment and a local environment, and was able to fix it by changing ideal to credit card.

Magento version: 2.3.4
PHP version: 7.3.19

Please fix this as soon as possible, thanks in advance.

Kind Regards,
Wouter Folkertsma

**Comment by paul-stolk-webdiensten** [comment]
Same here. I was able to solve it with:

````
if(window.checkoutConfig.payment.buckaroo.ideal){this.selectionType=window.checkoutConfig.payment.buckaroo.ideal.selectionType;}
````

But it would make more sense to use creditcard. 

This is about https://github.com/buckaroo-it/Magento2/blob/master/view/frontend/web/js/view/payment/method-renderer/creditcard.js#L78

**Comment by Buckaroo-Rens** [comment]
@paul-stolk-webdiensten Thank you for the information. We will try to research this a.s.a.p. Did you test on Buckaroo v1.26.0?

**Comment by paul-stolk-webdiensten** [comment]
> @paul-stolk-webdiensten Thank you for the information. We will try to research this a.s.a.p. Did you test on Buckaroo v1.26.0?

Buckaroo 1.26.0, Magento ver. 2.3.3-p1.

**Comment by seegrand** [comment]
This is the composer patch we used to fix it temporarily:

```
Buckaroo Github issue: https://github.com/buckaroo-it/Magento2/issues/56
===================================================================
--- /vendor/buckaroo/magento2/view/frontend/web/js/view/payment/method-renderer/creditcard.js
+++ /vendor/buckaroo/magento2/view/frontend/web/js/view/payment/method-renderer/creditcard.js
@@ -75,7 +75,11 @@
                     }
                     this.creditcards = ko.observableArray(filteredCards);

-                    this.selectionType  = window.checkoutConfig.payment.buckaroo.ideal.selectionType;
+                    if (window.checkoutConfig.payment.buckaroo.ideal){
+                        this.selectionType = window.checkoutConfig.payment.buckaroo.ideal.selectionType;
+                    } else {
+                        this.selectionType = "1";
+                    }

                     /**
                      * observe radio buttons
```

It hardcodes the selectionType to radio buttons (use `"2"` for dropdown) because when the iDeal payment method is disable, this object is not available in the creditcard conifg. But hey, it's something.

**Comment by Buckaroo-Rens** [comment]
The issue is fixed from Buckaroo version 1.28.0
